File: WEB2BEKEND/WEB2BEKEND/Models/SafetyDocument.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace WEB2BEKEND.Models
{
  public class SafetyDocument
  {
    [Key]
    public int Id { get; set; }
    public string Type { get; set; }
    public int SwitchingPlan { get; set; }
    public string Status { get; set; }
    public string Creator { get; set; }
    public string Crew { get; set; }
    public string Details { get; set; }
    public string Notes { get; set; }
    public string Phone { get; set; }
    public string State { get; set; }
    public string Date { get; set; }
    public bool OperationsComplited { get; set; }
    public bool TagsRemoved { get; set; }
    public bool GroundingRemoved { get; set; }
    public bool ReadyForService { get; set; }

    public ICollection<IncidentElement> Elements { get; set; }
    public ICollection<HistorySafeDocs> History { get; set; }
    public ICollection<Multimedia> Multimedia { get; set; }

    //public int Version { get; set; }
  }
}

File: src/app/entities/safetydoc/safetydocument.ts
import { HistorySafeDoc } from "../HistorySafeDoc";
import { Multimedia } from "../multimedia";
import { HistoryState } from "../spHistoryState";
import { Element } from "../element/element"

export class SafetyDocument {
    //basic
    id : number = 0;
    type : string = '';
    switchingPlan : number = 0;
    status : string = 'Drafted';
    creator : string = '';
    crew : string = '';
    details : string = '';
    notes : string = '';
    phone : string = '';
    state : string = 'Drafted';
    history : HistorySafeDoc[] = [];
    date : string = ''
    elements : Element[] = [];
    multimedia : Multimedia[] = [];
    operationsComplited : boolean = false;
    tagsRemoved : boolean = false;
    groundingRemoved : boolean = false;
    readyForService : boolean = false;
    //version : number = 0;
    constructor(id? : number, type? : string, plan? : number, creator? : string, crew? : string,
        details? : string, notes? : string, phone? : string, date? : string){
        if(id && type && plan && creator && crew && details && notes && phone && date){
            this.id = id;
            this.type = type;
            this.switchingPlan = plan;
            this.creator = creator;
            this.crew = crew;
            this.details = details;
            this.notes = notes;
            this.phone = phone;
            this.date = date;
            this.status = "Drafted";
        }
    }

}
